### PR TITLE
Merge branch 'feature_enable_secret_manager' into mainline

### PR DESCRIPTION
**Pull request by horsmand**
There were no merge conflicts attempting to merge the feature branch code into mainline.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by jusiskin — APPROVED** [review]

File: packages/aws-rfdk/lib/deadline/lib/render-queue.ts
Comment (jericht) [inline packages/aws-rfdk/lib/deadline/lib/render-queue.ts:265]: We'll need to update this to Deadline 10.1.19. Should we update this in this PR? or maybe we can just make a new PR targeting `mainline` once the Deadline version we need is released. What do you think?

**Review by jericht — APPROVED** [review]
LGTM! I left a comment so we don't forget to change the min version in the `RenderQueue`, that can be done either here or in a separate PR in mainline, but not holding this PR back for it.

File: packages/aws-rfdk/lib/deadline/lib/render-queue.ts
Comment (horsmand) [reply packages/aws-rfdk/lib/deadline/lib/render-queue.ts:265]: This has been completed in https://github.com/aws/aws-rfdk/pull/573 and is waiting for Deadline 10.1.19 to be released before we merge it.

**Pull request merged**
